src/sentry/stacktraces.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import
+
+import logging
+
+from sentry.models import Project
+from sentry.utils import metrics
+from sentry.utils.safe import safe_execute
+from collections import namedtuple
+
+
+logger = logging.getLogger(__name__)
+
+
+StacktraceInfo = namedtuple('StacktraceInfo', [
+    'stacktrace', 'container', 'platforms'])
+
+
+class StacktraceProcessor(object):
+
+    def __init__(self, data, stacktrace_infos, project=None):
+        self.data = data
+        self.stacktrace_infos = stacktrace_infos
+        if project is None:
+            project = Project.objects.get_from_cache(id=data['project'])
+        self.project = project
+
+    def close(self):
+        pass
+
+    def preprocess_related_data(self):
+        return False
+
+    def get_effective_platform(self, frame):
+        return frame.get('platform') or self.data['platform']
+
+    def process_frame(self, frame):
+        pass
+
+
+def find_stacktraces_in_data(data):
+    """Finds all stracktraces in a given data blob and returns it
+    together with some meta information.
+    """
+    rv = []
+
+    def _report_stack(stacktrace, container):
+        platforms = set()
+        for frame in stacktrace.get('frames') or ():
+            platforms.add(frame.get('platform') or data['platform'])
+        rv.append(StacktraceInfo(
+            stacktrace=stacktrace,
+            container=container,
+            platforms=platforms
+        ))
+
+    exc_container = data.get('sentry.interfaces.Exception')
+    if exc_container:
+        for exc in exc_container['values']:
+            stacktrace = exc.get('stacktrace')
+            if stacktrace:
+                _report_stack(stacktrace, exc)
+
+    stacktrace = data.get('sentry.interfaces.Stacktrace')
+    if stacktrace:
+        _report_stack(stacktrace, None)
+
+    threads = data.get('threads')
+    if threads:
+        for thread in threads['values']:
+            stacktrace = thread.get('stacktrace')
+            if stacktrace:
+                _report_stack(stacktrace, thread)
+
+    return rv
+
+
+def should_process_for_stacktraces(data):
+    from sentry.plugins import plugins
+    infos = find_stacktraces_in_data(data)
+    platforms = set()
+    for info in infos:
+        platforms.update(info.platforms or ())
+    for plugin in plugins.all(version=2):
+        processors = safe_execute(plugin.get_stacktrace_processors,
+                                  data=data, stacktrace_infos=infos,
+                                  platforms=platforms,
+                                  _with_transaction=False)
+        if processors:
+            return True
+    return False
+
+
+def get_processors_for_stacktraces(data, infos):
```

Comment:
Nitpicky comment, but there seems to be a decent amount of duplicate code between this and `should_process_for_stacktraces`.